apps/teams-test-app/src/components/utils/ApiWithTextInput.tsx

```diff
@@ -34,8 +40,18 @@ export const ApiWithTextInput = <T extends unknown>(props: ApiWithTextInputProps
       } else {
         const { validateInput, submit } = onClick;
         validateInput(partialInput);
-        const result = await submit(partialInput as T, setResult);
-        setResult(result);
+        if (typeof submit === 'function') {
+          const result = await submit(partialInput as T, setResult);
```

Comment:
can you defined `const input = partialInput as T` once before the `if`/`else` so the cast is not in 3 different places?

Reply:
Whoops, good catch! Making this change now